chat-plugins/ttt.js

```diff
@@ -0,0 +1,371 @@
+		game = tttgames[tttplayers[target.userid]];
+		if (game.p2.userid !== user.userid) return this.sendReply(target.name + ' has not challenged you to a game of Tic-Tac-Toe.');
+		tttplayers[user.userid] = tttplayers[target.userid];
+		game.accept();
+	},
+
+	dec: 'decline',
+	decline: function (target, room, user, connection, cmd) {
+		if (!target || !target.trim()) return this.sendReply('|html|/ttt ' + cmd + ' <em>User</em> - Declines a Tic-Tac-Toe challenge from a user.');
+		var targetUser = (Users.get(target) ? Users.get(target).name : target);
+		target = Users.get(target);
+		if (!target || !target.connected) return this.sendReply('User ' + targetUser + ' is offline.');
+		if (user.userid === target.userid) return this.sendReply('You can\'t use this command on yourself.');
+		var game = tttgames[tttplayers[toId(targetUser)]];
+		if (!(target.userid in tttplayers) || !game.checkPlayer(target)) return this.sendReply(target + ' has not challenged you to a game of Tic-Tac-Toe.');
+		if (game.checkPlayer(target) && game.phase == 'started') return this.sendReply('You are playing with ' + game.checkPlayer(user) + ' right now. If you want to end the game, use /ttt end.');
+
+		if (Users.get(target) && Users.get(target).connected) Users.get(target).send('|pm|' + user.getIdentity() + '|' + Users.get(target).getIdentity() + '|/error Your Tic-Tac-Toe request was declined.');
+		user.send('|pm|' + Users.get(target) + '|' + user.getIdentity() + '|/error You have declined the game request.');
+
+		game.end();
+	},
+
+	mark: 'markbox',
+	markbox: function (target, room, user, connection, cmd) {
+		if (!(user.userid in tttplayers)) return this.sendReply('You aren\'t playing a game of Tic-Tac-Toe right now.');
+		var game = tttgames[tttplayers[user.userid]];
+		if (game.phase === 'waiting') return this.sendReply('The request has not been accepted yet. You can only use this command in an active game.');
+		game.markBox(user, target);
+	},
+
+	update: 'see',
+	view: 'see',
+	show: 'see',
+	see: function (target, room, user) {
+		if (!(user.userid in tttplayers)) return this.sendReply('You aren\'t playing a game of Tic-Tac-Toe right now.');
+		var game = tttgames[tttplayers[user.userid]];
+		if (game.phase === 'waiting') return this.sendReply('Your request has not been accepted yet. You can only use this command in an active game.');
+		game.update();
+	},
+
+	exit: 'end',
+	leave: 'end',
+	end: function (target, room, user) {
+		if (!(user.userid in tttplayers)) return this.sendReply('You aren\'t playing a game of Tic-Tac-Toe right now.');
+		var game = tttgames[tttplayers[user.userid]];
+		if (game.phase === 'waiting') game.end('The request was withdrawn.');
+		else game.end(user.name + ' has decided to leave the game midway.');
+	}
+};
+
+exports.commands = {
+	ttt: 'tictactoe',
+	tictactoe: cmds,
+	tttend: 'endttt',
+	endttt: cmds.end,
+
+	//PM command override
+	pm: 'msg',
```

Comment:
ohh! Like it is when I Challenge you or some1 else for the game the challenges strike into the pm's